### PR TITLE
Preferences/settings/features cleanup [draft]

### PR DESCRIPTION
**Pull request by bvaughn**
## [Loom overview](https://www.loom.com/share/234d2f38a30b4233a33de4e8caa54508)

Remove the legacy Mozilla preferences service and replace it with a strongly-typed system preferences API that synchronizes to both `localStorage` and GraphQL.

Code for managing user data– whether `localStorage`, `IndexedDB`, or GraphQL– now lives in `shared/user-data`:
```
packages/shared/user-data
├── GraphQL
│   ├── UserData.ts
│   ├── config.ts
│   ├── constants.ts
│   ├── queries.ts
│   ├── types.ts
│   └── useGraphQLUserData.ts
├── IndexedDB
│   ├── cache.ts
│   ├── config.ts
│   ├── types.ts
│   ├── useIndexedDBUserData.ts
│   └── utils.ts
└── LocalStorage
    ├── config.ts
    ├── types.ts
    ├── useLocalStorageUserData.ts
    └── utils.ts
```

This PR also adds some documentation about how to choose which of the above services to manage user data. That documentation can be found here: [packages/shared/user-data/README.md](https://github.com/replayio/devtools/blob/FE-1483/packages/shared/user-data/README.md)

### CODE REVIEW

File: pages/browser/choose-role.tsx [hunk 1]
```diff
@@ -1,34 +1,22 @@
...
-      .catch(e => {
-        sendTelemetryEvent("DevtoolsGraphQLError", {
-          source: "updateUserSetting",
-          role,
-          message: e,
-        });
-
-        router.push("/");
-      });
+    await preferences.set("role", role);
```
Comment (jasonLaster) [inline pages/browser/choose-role.tsx:18]: 😍

File: jest.config.js [hunk 2]
```diff
@@ -52,10 +52,20 @@ module.exports = {
     // Use babel-jest to transpile tests with the next/babel preset
     // https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object
     "^.+\\.(js|jsx|ts|tsx)$": [
-      "babel-jest",
+      "@swc/jest",
       {
-        presets: ["next/babel"],
```
Comment (Andarist) [inline jest.config.js:57]: Eh, so it seems that Next is vendoring all Babel transforms and even the canary version of Next is still using old ones that do not support `satisfies`. It's virtually impossible for you to work around this on your own - unless you'd stop using the `next/babel` and just "inline" what they are using there but this time referring to the "real" packages.

I opened a PR to fix this: https://github.com/vercel/next.js/pull/51962

Comment (Andarist) [reply jest.config.js:57]: Note that switching to `@swc/jest` is the right call to do at some point anyway so this is as good time as ever :P

**Comment by bvaughn** [comment]
Need to split this into several separate PRs:
- [ ] Delete unused preferences
- [ ] Update dependencies
- [ ] Renamed `.ts` to `.tsx` file

**Comment by bvaughn** [comment]
I think the Delta tests failures are not specific to this branch, based on the two most recent runs from Jon's PR ([here](https://app.replay.io/team/dzpmMDZkNWI5OS05MWYzLTRhYTktYTYyNC0zMDRjYjJlYjBlYzk=/runs/eb4d6d72-f6f1-4bf7-a8af-6c869e937c2c) and [here](https://app.replay.io/team/dzpmMDZkNWI5OS05MWYzLTRhYTktYTYyNC0zMDRjYjJlYjBlYzk=/runs/128351f2-6524-4b0c-8967-23fbf5d0addc).

**Comment by bvaughn** [comment]
All tests are passing now. Woo hoo.

**Pull request merged**